### PR TITLE
Feature king movement

### LINKED ISSUE #14: Implementation for KING piece movements
This can be implemented after issue #12 is solved.

### PR DESCRIPTION
**Pull request by myogh**
King movement has been implemented. Requesting your reviews. @mlsdpk @Awthura @aungpaing98

### CODE REVIEW

File: checkers/Board.py [hunk 1]
```diff
@@ -206,6 +410,22 @@ def is_free_space(self, row, col):
     def is_same_player(self, row, col, player_turn):
         return self.grid[row][col].player == player_turn
 
+    def make_kings_if_any(self, player_turn):
```
Comment (mlsdpk) [inline checkers/Board.py:413]: Implementation of this function is great, however, do we need to loop the grid for checking the piece is king or not? I think maybe we can check in the _move_piece_ function (after this line `self.grid[selected_row][selected_col].position = (selected_row, selected_col)`). Any thoughts? @aungmcs @Awthura @aungpaing98

File: checkers/Board.py [hunk 2]
```diff
@@ -267,14 +491,15 @@ def render(self, screen, player_turn):
...
+                    
+                    if self.grid[row][col].player == -1:
                         color = self.colors['BLACK']
 
                     pygame.draw.circle(
                         screen, color,
                         (col * self.grid_size + self.grid_size / 2,
                          row * self.grid_size + self.grid_size / 2),
-                        self.grid_size / 3)
+                        self.grid_size / 3, draw_top_right = self.grid[row][col].status == 'king')
```
Comment (mlsdpk) [inline checkers/Board.py:502]: I assume this is not the final king rendering right? Any ideas for king rendering?

File: checkers/Piece.py [hunk 3]
```diff
@@ -4,5 +4,6 @@ def __init__(self, row, col, player):
 
         self.position = (row, col)
         self.player = player
+        self.status = 'man'
```
Comment (mlsdpk) [inline checkers/Piece.py:7]: Adding this attribute is nice.

File: checkers/Board.py [hunk 4]
```diff
@@ -195,7 +245,161 @@ def find_valid_moves_recursive(self, player_turn):
                                 self.piece_set.add(
                                     (new_right_grid_row, new_right_grid_col))
 
-            return self.find_valid_moves_recursive(player_turn)
+            return self.find_man_valid_moves(player_turn)
+
+    def find_king_valid_moves(self, player_turn):
```
Comment (mlsdpk) [inline checkers/Board.py:250]: Not sure about this function yet, if it's possible, combing both man and king functions might be nice but need to find a way to shorten the code. Need to discuss about this.

**Review by mlsdpk — COMMENTED** [review]
I have reviewed the codes, we might need to discuss king implementation further more, my idea is to combine man and king implementations together (also need to discuss the king rules).

File: checkers/Board.py (hunk 2 above)
Comment (myogh) [reply checkers/Board.py:502]: Yes. It's a temporary implementation.

**Pull request merged**